server/cli/src/cli.rs

```diff
@@ -524,11 +539,77 @@ async fn main() -> anyhow::Result<()> {
         Action::GeneratePluginBundle(arguments) => {
             generate_plugin_bundle(arguments)?;
         }
+        Action::ShowReport { path, config } => {
+            let report_data: ReportData = generate_report_data(&path)?;
+
+            let report_json =
+                serde_json::to_value(report_data.template).expect("fail to convert report to json");
+
+            let test_config_path = if let Some(config) = config {
+                config
+            } else {
+                Path::new("reports").to_path_buf()
+            };
+
+            let test_config_file = fs::File::open(test_config_path.join("test-config.json"))
+                .map_err(|e| {
+                    ReportError::CannotOpenTestConfigFile(test_config_path.to_path_buf(), e)
+                })?;
+            let test_config: TestConfig =
+                serde_json::from_reader(test_config_file).map_err(|e| {
+                    ReportError::CannotReadTestConfigFile(test_config_path.clone().to_path_buf(), e)
+                })?;
+
+            let config = Config {
+                url: test_config.url,
+                username: test_config.username,
+                password: test_config.password,
+            };
+
+            let output_name = format!("{}.html", test_config.output_filename.clone());
+
+            let report_generate_data = ReportGenerateData {
+                report: report_json,
+                config: config,
+                store_id: Some(test_config.store_id),
+                store_name: None,
+                output_filename: Some(output_name.clone()),
+                format: Format::Html,
+                data_id: Some(test_config.data_id),
+                arguments: Some(test_config.arguments),
+            };
+
+            let res = spawn_blocking(|| generate_report_inner(report_generate_data)).await?;
```

Comment:
I don't quite understand how the spawn_blocking works, so unsure if this is the correct approach.